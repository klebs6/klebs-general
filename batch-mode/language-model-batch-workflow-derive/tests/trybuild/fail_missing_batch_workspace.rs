// ---------------- [ File: tests/trybuild/fail_missing_batch_workspace.rs ]
// -------------------------------------------------------------------------------------------
// [ File: tests/trybuild/fail_missing_batch_workspace.rs ]
//
// Similarly, no `#[batch_workspace]` => compile-fail.

use language_model_batch_workflow_derive::LanguageModelBatchWorkflow;
use batch_mode_batch_workflow::*;

// "fail_missing_batch_workspace.rs"
struct MissingWorkspace {
    #[batch_client]
    client: std::sync::Arc<OpenAIClientHandle>,

    #[expected_content_type]
    ect: ExpectedContentType,

    #[model_type]
    mt: LanguageModelType,
}

struct MyErr;

// Expect an error: “Missing required `#[batch_workspace]`.”
fn main() {}
