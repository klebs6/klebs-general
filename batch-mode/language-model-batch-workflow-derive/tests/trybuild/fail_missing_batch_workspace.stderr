error: cannot find attribute `batch_client` in this scope
  --> tests/trybuild/fail_missing_batch_workspace.rs:10:7
   |
10 |     #[batch_client]
   |       ^^^^^^^^^^^^

error: cannot find attribute `expected_content_type` in this scope
  --> tests/trybuild/fail_missing_batch_workspace.rs:13:7
   |
13 |     #[expected_content_type]
   |       ^^^^^^^^^^^^^^^^^^^^^

error: cannot find attribute `model_type` in this scope
  --> tests/trybuild/fail_missing_batch_workspace.rs:16:7
   |
16 |     #[model_type]
   |       ^^^^^^^^^^
   |
note: `model_type` is imported here, but it is a module, not an attribute
  --> tests/trybuild/fail_missing_batch_workspace.rs:6:5
   |
6  | use batch_mode_batch_workflow::*;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused import: `language_model_batch_workflow_derive::LanguageModelBatchWorkflow`
 --> tests/trybuild/fail_missing_batch_workspace.rs:5:5
  |
5 | use language_model_batch_workflow_derive::LanguageModelBatchWorkflow;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
